koku/api/utils.py

```diff
@@ -450,35 +450,60 @@ def materialized_view_month_start(dh=DateHelper()):
     return dh.this_month_start - relativedelta(months=settings.RETAIN_NUM_MONTHS - 1)
 
 
+def get_split_date_string(date_str):
+    """Return split date string"""
+
+    if not isinstance(date_str, str):
+        return date_str
+
+    split_at_char = ["T", " "]
+    for char in split_at_char:
+        if char.lower() in date_str.lower():
+            return date_str.split(char)[0]
+    return date_str
+
+
 def get_months_in_date_range(report=None, start=None, end=None, invoice_month=None):
```

Comment:
There is a lot of type conversion internal to this method. The internal workings would be much cleaner if the inputs, which seem to be `str` based on how they are treated, were converted to `datetime` objects early, and only converted to strings at the `return`.

This will also eliminate the need for `get_split_date_string`.